### PR TITLE
add-github-actions

### PR DESCRIPTION
**Pull request by nabim777**
## Description
Here ,setup of GitHub actions workflow is added to test only

## Issue
https://github.com/JankariTech/expected-failures-updater/issues/11

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        run: go version
+
+      - name: Install bats
+        run: |
+          sudo apt-get install bats
+
+      - name: bats run
+        working-directory: ./
+        run: |
+          bats tests/test_scan_command.bats
```
Comment (amrita-shrestha) [inline .github/workflows/test.yml]:
```suggestion
      - name: bats run
        working-directory: ./tests
        run: |
          bats test_scan_command.bats
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.15'
+
+      - name: Display Go version
+        run: go version
+
+      - name: Install bats
+        run: |
+          sudo apt-get install bats
```
Comment (amrita-shrestha) [inline .github/workflows/test.yml]: name: Setup BATS
uses: mig4/setup-bats@v1
with:
bats-version: 1.2.1

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          go-version: '1.15'
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.54
+          run: golangci-lint
+
+      - name: Display Go version
+        run: go version
```
Comment (SagarGi) [inline .github/workflows/test.yml]:
```suggestion
      - name: Display Go version
        run: go version
```
is this step necessary to display go version? i mean we already know go verison.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  go:
```
Comment (SagarGi) [inline .github/workflows/test.yml]: A suitable job name would be appropriate instead of `go` may be?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+name: CI
```
Comment (saw-jan) [inline .github/workflows/test.yml]:
```suggestion
name: tests
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Setup BATS
+        uses: mig4/setup-bats@v1
+        with:
+          bats-version: 1.2.1
+
+      - name: Run bats test
+        working-directory: ./
+        run: |
+          bats tests/test_scan_command.bats
```
Comment (saw-jan) [inline .github/workflows/test.yml]: I don't see build step. So, tests can run without build?

File: .github/workflows/test.yml (hunk 3 above)
Comment (nabim777) [reply .github/workflows/test.yml]: Ok i will remove➖ it.

**Review by individual-it — COMMENTED** [review]
should we fix the linting issues in this PR or in a seperate?

**Comment by nabim777** [comment]
> should we fix the linting issues in this PR or in a seperate?

We can fix in this PR

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+name: tests
```
Comment (SwikritiT) [inline .github/workflows/tests.yml]: can we also change the name of the file to `ci.yml`? 
```suggestion
name: CI
```

File: readme.md [hunk 8]
```diff
@@ -58,3 +58,12 @@ go run main.go <command>
...
     ```
 
+### Run test
+- Install Bats (Bash Automated Testing System) on your system using the following command:
+    ```bash
+    sudo apt install bats
+    ```      
+- After installing Bats, you can run your tests locally using the following command:
+    ```bash 
+    bats tests/test_scan_command.bats
```
Comment (SwikritiT) [inline readme.md]: maybe better to not hard-code the filename incase we add other tests as well?
```suggestion
    bats tests/<path-to-the-test-file>  
```

File: readme.md [hunk 9]
```diff
@@ -58,3 +58,12 @@ go run main.go <command>
     FEATURES_PATH=<path_to_feature_files> EXPECTED_FAILURES_DIR=<path_to_expected_failures> go run main.go shift
     ```
 
+### Run test
+- Install Bats (Bash Automated Testing System) on your system using the following command:
+    ```bash
+    sudo apt install bats
+    ```      
+- After installing Bats, you can run your tests locally using the following command:
```
Comment (SwikritiT) [inline readme.md]:
```suggestion
- After installing Bats, you can run your tests locally from the base directory of this project using the following command:
```

File: readme.md (hunk 8 above)
Comment (nabim777) [reply readme.md]: Can it be done this way?
`bats tests/*.bats`

File: .github/workflows/CI.yml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Setup BATS
+        uses: mig4/setup-bats@v1
+        with:
+          bats-version: 1.2.1
+
+      - name: Run bats test
+        working-directory: ./
+        run: |
+          bats tests/*.bats
```
Comment (SagarGi) [inline .github/workflows/CI.yml]:
```suggestion
        run: bats tests/*.bats
```

File: readme.md (hunk 8 above)
Comment (phil-davis) [reply readme.md]: yes, that will work and in the future a developer can run all the bats tests without needing to know what the filename(s) are.
```
phil@phil-Inspiron-5468:~/git/JankariTech/expected-failures-updater$ bats tests/*.bats
 ✓ detect removed scenarios
 ✓ detect new added scenarios

2 tests, 0 failures
```

File: readme.md [hunk 11]
```diff
@@ -58,3 +58,12 @@ go run main.go <command>
...
     ```
 
+### Run test
+- Install Bats (Bash Automated Testing System) on your system using the following command:
+    ```bash
+    sudo apt install bats
+    ```      
+- After installing Bats, you can run your tests locally from the base directory of this project using the following command:
+    ```bash 
+    bats tests/test_scan_command.bats
```
Comment (phil-davis) [inline readme.md]:
```suggestion
    bats tests/*.bats  
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**